Makefile

```diff
@@ -51,9 +53,70 @@ install-systemd:
 	install -d $(DEST_SYSTEMD)
 	install -m 0644 $(SRCS_SYSTEMD) $(DEST_SYSTEMD)
 
+install-current-version-file:
+	install -d $$(dirname $(DEST_CURRENT_VERSION_FILE))
+	git rev-parse --short HEAD > $(DEST_CURRENT_VERSION_FILE)
+
 # target: uninstall - Uninstall ALL files from the install targets.
 uninstall:
 	@for file in $(INSTALLED_FILES); do \
 			echo $(RM) $$file; \
 			$(RM) $$file; \
 	done
+
+# target: upgrade - Upgrade installation.
+# .ONESHELL runs the script in a single shell instead of a shell per line; so no semicolons, nor backslashes required :)
+.ONESHELL:
+upgrade:
```

Comment:
As this is quite a lot of logic, I think it would do better in a script of its own, with the needed inputs as arguments to it.